app/api/ada/lib/state-fetch/remoteFetcher.js

```diff
@@ -43,6 +43,10 @@ export class RemoteFetcher implements IFetcher {
         method: 'post',
         data: {
           addresses: body.addresses
+        },
+        headers: {
+          'x-yoroi-version': window.yoroi.stores.profile.lastLaunchVersion,
```

Comment:
Using `window` is dangerous since it can easily break on refactoring so I don't recommend it. Probably it's best if instead `RemoteFetcher` takes a lambda as a constructor argument.
Now from the `StateFetcherStore` you can access `stores.profile` safely from inside the `setup` function. Use this to construct a lambda you pass into the the `RemoteFetcher` fetcher.

Probably if you do that, you will get some Flow complains that now the `fetcher` variable in `StateFetchStore` can be undefined. You can solve this by either
A) Creating a new DummyFetcher that just throws an exception on any function call and use this as the default value (kind of an ugly hack)
B) Create a function called `getFetcher` in `StateFetchStore` and make the function throw if the fetcher is uninitialized. (probably  much cleaner)